src/plugins/creditor/credNodeStore.test.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import {type CreditorNode} from "./creditorNode";
+import Store, {getDetails, toNodeEntry} from "./credNodeStore";
+import {random as randomUuid} from "../../util/uuid";
+
+describe("plugins/creditor/credNodeStore", () => {
+  const tagId1 = randomUuid();
+  const tagId2 = randomUuid();
+  const baseNode: CreditorNode = {
+    id: randomUuid(),
+    tags: [tagId1, tagId2],
+    title: "Implementing credNodes",
+    description: "creating credNodes for use in the creditor",
+    graphTimestamp: new Date("1/1/2000").getTime(),
+    createdAt: new Date("12/1/2000").getTime(),
+    mint: 0,
+    parent: null,
+  };
+
+  const childNode: CreditorNode = {
+    id: randomUuid(),
+    tags: [tagId2],
+    title: "Creating credNode type",
+    description: "specifying credNodes to test the store",
+    graphTimestamp: new Date("1/1/2000").getTime(),
+    createdAt: new Date("11/1/2000").getTime(),
+    mint: 20,
+    parent: baseNode.id,
+  };
+
+  describe("adding nodes to store", () => {
+    it("requires nodes to have expected members", () => {
+      const store = new Store();
+      const thunk = (badNode: Object) => () => store.set(badNode);
+
+      const badNodes = [
+        [{}, "Creditor Node UUID required"],
+        [{id: randomUuid()}, "`tags` must be an array"],
+        [
+          {id: randomUuid(), tags: []},
+          "`title` and `description` are required in Cred Nodes",
+        ],
+        [
+          {
+            id: randomUuid(),
+            tags: [],
+            title: "bad Node",
+            description: "badDescription",
+            createdAt: 0,
+            graphTimestamp: 0,
+          },
+          "`createdAt` and `graphTimestamp` values must be non-zero",
+        ],
+      ];
+      badNodes.forEach(([node, errorMsg]) =>
+        expect(thunk(node)).toThrowError(errorMsg)
+      );
+    });
+    it("can set and get valid nodes", () => {
+      const store = new Store();
+      const nextStore = store.set(baseNode);
+      expect(nextStore).toEqual(store);
+      const returnedBase = store.get(baseNode.id);
+      expect(returnedBase).toEqual(baseNode);
+    });
+    it("returns null for nonexistent nodes", () => {
+      const store = new Store();
+      const badId = randomUuid();
+      const result = store.get(badId);
+      expect(result).toBeEmpty;
+    });
```

Comment:
This test does not do what it says on the tin. The store actually returns `undefined`, but the test passes.

A correct implementation of the test (which would fail) would be:
`expect(store.get(badId)).toBe(null)`

It's really dumb that JS has null and undefined as distinct values, but it does. As a heuristic, when we're writing our code we almost always prefer using explicit null, except in the case where we are implementing `Map.get` esque-semantics, in which case we return `undefined` to be consistent with the language primitives.